SpatialGDK/Source/SpatialGDKTests/SpatialGDK/RPCService/RPCServiceTest.cpp

```diff
@@ -0,0 +1,568 @@
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ClientReliable, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::NoRingBufferAuthority;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_client_endpoint_WHEN_push_client_unreliable_rpcs_to_the_service_THEN_check_no_buffer_authority)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, CLIENT_AUTH);
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ClientUnreliable, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::NoRingBufferAuthority;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_client_endpoint_WHEN_push_server_reliable_rpcs_to_the_service_THEN_check_success)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, CLIENT_AUTH);
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ServerReliable, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::Success;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_client_endpoint_WHEN_push_server_unreliable_rpcs_to_the_service_THEN_check_success)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, CLIENT_AUTH);
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ServerUnreliable, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::Success;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_client_endpoint_WHEN_push_multicast_rpcs_to_the_service_THEN_check_success)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, CLIENT_AUTH);
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::NetMulticast, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::NoRingBufferAuthority;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_server_endpoint_WHEN_push_multicast_rpcs_to_the_service_THEN_check_success)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, SERVER_AUTH);
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::NetMulticast, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::Success;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_server_and_client_endpoint_WHEN_push_rpcs_to_the_service_THEN_check_push_results)
```

Comment:
This test name isn't as clear as the others. After the `THEN`, what are we expecting to happen? `has_ack_authority` ?

Reply:
Done